libs/message-coder/src/uint16_coder.ts

```diff
@@ -0,0 +1,48 @@
+import { Buffer } from 'buffer';
+import {
+  BitLength,
+  BitOffset,
+  Coder,
+  DecodeResult,
+  EncodeResult,
+  mapResult,
+  Uint16,
+} from './types';
+import { UintCoder } from './uint_coder';
+
+/**
+ * Coder for a uint16, aka a 16-bit unsigned integer.
+ */
+export class Uint16Coder extends UintCoder {
+  bitLength(): BitLength {
+    return 16;
+  }
+
+  encodeInto(
+    value: Uint16,
+    buffer: Buffer,
+    bitOffset: BitOffset
+  ): EncodeResult {
+    return mapResult(this.validateValue(value), () =>
+      this.encodeUsing(buffer, bitOffset, (byteOffset) =>
+        buffer.writeUInt16LE(value, byteOffset)
```

Comment:
FWIW, the paper handler driver sometimes takes input in `UInt16` little endian and sometime big endian, it's not consistent.

Reply:
Good to know. We can add support for big endian as needed.